src/outputs/restart.hpp

```diff
@@ -83,263 +63,63 @@ class RestartReader {
     }
   };
 
-  SparseInfo GetSparseInfo() const;
-
-  // Return output format version number. Return -1 if not existent.
-  int GetOutputFormatVersion() const;
+  [[nodiscard]] virtual SparseInfo GetSparseInfo() const = 0;
 
- private:
-  struct DatasetHandle {
-    hid_t type;
-    H5D dataset;
-    H5S dataspace;
-    int rank;
-    hsize_t count;
-    std::vector<hsize_t> dims;
+  struct MeshInfo {
+    int nbnew, nbdel, nbtotal, root_level, includes_ghost, n_ghost;
+    std::vector<std::string> bound_cond;
+    std::vector<int> block_size;
+    std::vector<Real> grid_dim;
+    std::vector<int64_t> lx123;
+    std::vector<int> level_gid_lid_cnghost_gflag; // what's this?!
```

Comment:
Do we want all of this stuff mashed together going forward? For the forest stuff, we would also have to add a tree index here.

Reply:
If you mean "all mesh related stuff" with "all", then I'd imagine so.
I separate "time" and "mesh" similar to the separate `SparseInfo` though I'm also open to suggestion for other ways.
The key issue was trying to resolve is that `GetAttr<T>` in a templated version could not be used in the abstract class anymore, so I introduced this interface so that all the `GetAttr<T>` calls are private to the derived class.

Reply:
Does this belong in restart, actually? Or should it be elsehwere?

Reply:
Do you mean to introduce a struct in `Mesh` that holds this info so it can be reused (similar to `SimTime`)?